src/core/context_processors.py

```diff
@@ -54,5 +54,22 @@ def sponsors(request):
     }
 
 
+def _get_sponsors_with_open_roles():
+    sponsor_set = set()
+    for sponsor in Sponsor.objects.order_by('level'):
+        for open_role in OpenRole.objects.order_by('sponsor'):
+            if open_role.sponsor.name == sponsor.name:
+                sponsor_set.add(sponsor)
```

Comment:
I remember that the query will be executed repeatedly in the for loop, and it will be better if it is changed to a code like the following.
```
sponsor_has_open_role = set(OpenRole.objects.values_list('sponsor', flat=True))
sponsor_set = Sponsor.objects.filter(id__in=sponsor_has_open_role).order_by('level')
````